netebpfext/net_ebpf_ext_sock_addr.c

```diff
@@ -814,9 +821,16 @@ net_ebpf_ext_sock_addr_register_providers()
 void
 net_ebpf_ext_sock_addr_unregister_providers()
 {
-    for (int i = 0; i < NET_EBPF_SOCK_ADDR_HOOK_PROVIDER_COUNT; i++)
-        net_ebpf_extension_hook_provider_unregister(_ebpf_sock_addr_hook_provider_context[i]);
-    net_ebpf_extension_program_info_provider_unregister(_ebpf_sock_addr_program_info_provider_context);
+    for (int i = 0; i < NET_EBPF_SOCK_ADDR_HOOK_PROVIDER_COUNT; i++) {
+        if (_ebpf_sock_addr_hook_provider_context[i]) {
+            net_ebpf_extension_hook_provider_unregister(_ebpf_sock_addr_hook_provider_context[i]);
+            _ebpf_sock_addr_hook_provider_context[i] = NULL;
```

Comment:
This is ok, but what about changing `net_ebpf_extension_hook_provider_unregister` to set the argument to NULL on return so it's automatic for all callers?